roman_kuprii/graph_generation_controller.hpp

```diff
@@ -24,34 +24,35 @@ class GraphGenerationController {
     explicit Worker(const GetJobCallback& get_job_callback)
         : get_job_callback_(get_job_callback){};
 
+    enum class State { Idle, Working, ShouldTerminate };
+
     void start();
     void stop();
 
-    bool should_terminate() { return finish_flag; }
+    ~Worker();
 
    private:
     std::thread thread_;
     GetJobCallback get_job_callback_;
-    bool finish_flag = false;
+    State state_flag_ = State::Idle;
```

Comment:
Будешь у каждой переменной добавлять суффикс `flag`?
```suggestion
    State state = State::Idle;
```

Reply:
`std::atomic`?